pkg/summarizer/summary.go

```diff
@@ -73,142 +70,131 @@ type gridReader func(ctx context.Context) (io.ReadCloser, time.Time, int64, erro
 // groupFinder returns the named group as well as reader for the grid state
 type groupFinder func(string) (*configpb.TestGroup, gridReader, error)
 
-// Update summary protos by reading the state protos defined in the config.
-//
-// Will use concurrency go routines to update dashboards in parallel.
-// Setting dashboard will limit update to this dashboard.
-// Will write summary proto when confirm is set.
-func Update(ctx context.Context, client gcs.ConditionalClient, mets *Metrics, configPath gcs.Path, concurrency int, dashboard, gridPathPrefix, summaryPathPrefix string, confirm bool) error {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	if concurrency < 1 {
-		return fmt.Errorf("concurrency must be positive, got: %d", concurrency)
-	}
-	cfg, err := config.ReadGCS(ctx, client, configPath)
+func fetchConfig(ctx context.Context, client gcs.ConditionalClient, configPath gcs.Path, dashboard string) (*configpb.Configuration, *storage.ReaderObjectAttrs, error) {
+	r, attrs, err := client.Open(ctx, configPath)
 	if err != nil {
-		return fmt.Errorf("Failed to read config: %w", err)
+		return nil, nil, fmt.Errorf("open: %w", err)
 	}
-	log := logrus.WithField("config", configPath)
-	log.WithField("dashboards", len(cfg.Dashboards)).Info("Updating dashboards")
-
-	dashboards := make(chan *configpb.Dashboard)
-	var wg sync.WaitGroup
 
-	var generations map[string]int64
+	cfg, err := config.Unmarshal(r)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
 
-	groupFinder := func(name string) (*configpb.TestGroup, gridReader, error) {
-		group := config.FindTestGroup(name, cfg)
-		if group == nil {
-			return nil, nil, nil
-		}
-		groupPath, err := configPath.ResolveReference(&url.URL{Path: path.Join(gridPathPrefix, name)})
-		if err != nil {
-			return group, nil, err
+	if dashboard != "" {
```

Comment:
There's an opportunity in this loop to cut out all test groups that aren't the one this dashboard displays.
If it's worth trimming the config to one dashboard, it's probably worth trimming the test groups too.

Reply:
We need to trim the dashboards otherwise we don't update them. 

The presence of extra testgroups is not a big deal right now. Since the logic to remove them is not straight forward so this seems cheaper and less error prone.